### PR TITLE
Re-merge Pyk into K

### PR DESCRIPTION
**Pull request by Baltoli**
This PR merges the Pyk library back into K such that it becomes a first-class citizen once again, rather than a separately-maintained library. The changes here are a full import of Pyk's history (cleaned so that file histories are preserved), and a series of small changes that set up CI and clean up the resulting issues from performing the merge.

> [!IMPORTANT]  
> This is a major piece of infrastructural change for our code, and it should be reviewed as carefully as is practical given the nature of the changes.

There are nearly 800 commits in Pyk's history, and so reviewing the whole history of this PR is not practical. The salient parts to review are the commits I have authored (beginning at [e5c3370](https://github.com/runtimeverification/k/pull/4164/commits/e5c33709eda58c4f87af798fdbbe7e7cc5a40654), and ending at [5caa31c](https://github.com/runtimeverification/k/pull/4164/commits/5caa31c39ab6d0a9cce7cdaa5bbe086895c8018f)); their histories are clean and should be easy to review.

Additionally, please take a look through the repository tree as it is on this branch, and check that the `pyk/` subtree looks reasonable to you (histories, etc. are all sensible). I would like @tothtamas28, @ehildenb and @F-WRunTime in particular to check this all over before merging. 

See https://github.com/runtimeverification/k/issues/4066#issuecomment-2044658383 for a blow-by-blow account of what I've done to produce this PR.

> [!CAUTION]
> We **must not** use the RV automerger for this PR; it will squash the Pyk history into a single commit (which isn't what we want). Instead, once this PR has been reviewed and approved, it should be merged using the GitHub user interface by an admin. Optionally, we can squash the non-pyk commits to keep the history outside the subdirectory smaller.

Once the PR has been reviewed and is ready to merge, I will take the following steps:
- Squash my changes down into a single commit, leaving the Pyk history separate
- Make sure that the branch is up to date and no other PRs are going to automerge during the merge of this PR
- Make sure that the release job is stable
- Manually merge this PR without squashing

There will be more cleanup of issues, PRs, CI and other ancillary parts once this PR merges, but these issues are off the critical path.

Closes: https://github.com/runtimeverification/k/issues/4066

### CODE REVIEW

File: .github/workflows/test-pr.yml [hunk 1]
```diff
@@ -102,6 +102,162 @@ jobs:
           if-no-files-found: error
           retention-days: 1
 
+  pyk-code-quality-checks:
```
Comment (tothtamas28) [inline .github/workflows/test-pr.yml:105]: Would it make sense to put these into a separate workflow? (If I know correctly triggers are per-workflow.)

In that case of course, the Debian package would have to be shared between the workflows.

File: pyk/.gitignore [hunk 2]
```diff
@@ -1,8 +1,4 @@
-/dist/
```
Comment (tothtamas28) [inline pyk/.gitignore]: If there's a way, it'd be nice to keep this modular:
* `.gitignore` does not apply in `pyk`
* `pyk/.gitignore` applies in `pyk`

File: pyk/pyproject.toml [hunk 3]
```diff
@@ -4,7 +4,7 @@ build-backend = "poetry.core.masonry.api"
 
 [tool.poetry]
 name = "pyk"
-version = "0.1.779"
+version = "6.3.0"
```
Comment (tothtamas28) [inline pyk/pyproject.toml]: There's also `K_VERSION` in `pyk/src/pyk/__init__.py`, which for now we can simply replace by `__version__`.

File: pyk/deps/k_release
Comment (tothtamas28) [inline pyk/deps/k_release:1]: It is probably safe now to remove this file.

**Review by tothtamas28 — APPROVED** [review]

**Review by F-WRunTime — APPROVED** [review]

**Pull request merged**